### PR TITLE
Add the RunOnce decorator node

### PR DESCRIPTION
**Pull request by galou**
This adds a RunOnce decorator node that will call its child' `executeTick()` only once. After the first time, it'll return the return value of child's `status()`.

Signed-off-by: Gaël Écorchard <gael.ecorchard@cvut.cz>

### CODE REVIEW

**Comment by facontidavide** [comment]
It is ok for me to merge this

File: include/behaviortree_cpp/decorators/run_once_node.h [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  if (already_ticked_)
+  {
+    return child_node_->status();
+  }
+  else
+  {
+    return child_node_->executeTick();
+  }
+  already_ticked_ = true;
```
Comment (facontidavide) [inline include/behaviortree_cpp/decorators/run_once_node.h:52]: I think this is what you want:

```suggestion
  else
  {
    already_ticked_ = true;
    return child_node_->executeTick();
  }
```

**Comment by galou** [comment]
I changed according to your wish but the functionality was not affected.

**Comment by facontidavide** [comment]
I am still thinking about what the desired behavior is for Asynchronous children.

May be this make more sense?

```c++
auto status = child_node_->executeTick();
if(status == NosteStatus::SUCCESS || status == NosteStatus::FAILURE) {
   already_ticked_ = true;
}
return status;
```

Otherwise we don't complete RUNNING children

**Comment by facontidavide** [comment]
no, not necessarily

**Comment by facontidavide** [comment]
Merged with some important improvements 

3efde1e795749f4d1d982ed3dc53c3f2745e39e9

**Comment by galou** [comment]
Nice, sorry for not having reacted on this.

**Pull request closed without merging**